From 73ea388abbbd46381469edc3dbf8543e8df2d0cc Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 13 Aug 2019 18:32:11 +0300
Subject: [PATCH] [SW-0] gaudi: fix compilation erroe

gaudi/asic.c: In function ‘hl_skip_pcie_init’:
gaudi/asic.c:301:9: error: suggest parentheses around operand of ‘!’ or
change ‘&’ to ‘&&’ or ‘!’ to ‘~’ [-Werror=parentheses]
  return !regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Change-Id: I00b17547434579f0ac2a5479f0aebad711b4c989
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index c760a0d..87785f1 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -298,7 +298,7 @@
 u32 hl_skip_pcie_init(void)
 {
 	/* Use DumpSel[11] to skip PCIe initialization (active low) */
-	return !regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
+	return !(regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19));
 }
 
 u64 hl_pll_get_cpu_cycles_per_usec(void)
